internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -121,25 +124,33 @@ func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, s c
 	}
 
 	for name, status := range in.DependsOn {
-		if !isValidStatus(status) {
-			return errInvalidDependsOnStatus
+		if ok, err := isValidStatus(status, name, s); !ok {
+			return err
 		}
-		if isEssential(name, s) && !isEssentialStatus(status) {
-			return errEssentialContainerStatus
+		if ok, err := isEssentialStatus(status, name, s); isEssential(name, s) && !ok {
+			return err
 		}
 	}
 
 	return nil
 }
 
-func isValidStatus(s string) bool {
-	for _, allowed := range dependsOnStatus {
-		if s == allowed {
-			return true
+func isValidStatus(status string, container string, c convertSidecarOpts) (bool, error) {
```

Comment:
Can we rename this to `isValidStatusForContainer`? It might be a bit more transparent what we're checking that way.

Reply:
Adjusted naming.